src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapQueryString;
+use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Exception\PartialDenormalizationException;
+use Symfony\Component\Serializer\Exception\UnsupportedFormatException;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class RequestPayloadValueResolver implements ValueResolverInterface
+{
+    private const DEFAULT_FORMAT = 'json';
+
+    /**
+     * @see \Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::DISABLE_TYPE_ENFORCEMENT
+     * @see DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS
+     */
+    private const CONTEXT_DENORMALIZE = [
+        'disable_type_enforcement' => true,
+        'collect_denormalization_errors' => true,
+    ];
+
+    /**
+     * @see DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS
+     */
+    private const CONTEXT_DESERIALIZE = [
+        'collect_denormalization_errors' => true,
+    ];
+
+    public function __construct(
+        private readonly SerializerInterface&DenormalizerInterface $serializer,
+        private readonly ?ValidatorInterface $validator,
+    ) {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $payloadMappers = [
+            MapQueryString::class => ['mapQueryString', Response::HTTP_NOT_FOUND],
```

Comment:
The triple dots closure syntax would improve readability and static analysis.
```suggestion
            MapQueryString::class => [$this->mapQueryString(...), Response::HTTP_NOT_FOUND],
```

Reply:
That'd create a closure for nothing, aka have a perf cost.